doc/man3/OSSL_CMP_CTX_new.pod

```diff
@@ -0,0 +1,686 @@
+                                        which is the default.
+            OSSL_CRMF_POPO_KEYENC     - decrypt the encrypted certificate
+                                        ("indirect method")
+
+        Note that a signature-based POPO can only produced if a private key
+        is provided as the newPkey or client pkey component of the CMP context.
+
+=item B<OSSL_CMP_OPT_DIGEST_ALGNID>
+
+        The digest algorithm NID to be used in RFC 4210's MSG_SIG_ALG,
+        if applicable used for message protection and Proof-of-Possession.
+        Default is SHA256.
+
+    OSSL_CMP_OPT_OWF_ALGNID
+        The digest algorithm NID to be used as one-way function (OWF)
+        in RFC 4210's MSG_MAC_ALG, if applicable used for message protection.
+        Default is SHA256.
+
+    OSSL_CMP_OPT_MAC_ALGNID
+        The MAC algorithm NID to be used in RFC 4210's MSG_MAC_ALG,
+        if applicable used for message protection. 
+        Default is HMAC-SHA1 as per RFC 4210.
+
+=item B<OSSL_CMP_OPT_REVOCATION_REASON>
+
+        The reason code to be included in a Revocation Request (RR);
+        values: 0..10 (RFC 5210, 5.3.1) or -1 for none (which is the default)
+
+=item B<OSSL_CMP_OPT_IMPLICITCONFIRM>
+
+        Request server to enable implicit confirm mode, where the client
+        does not need to send confirmation upon receiving the
+        certificate. If the server does not enable implicit confirmation
+        in the return message, then confirmation is sent anyway.
+
+=item B<OSSL_CMP_OPT_DISABLECONFIRM>
+
+        Do not confirm enrolled certificates, to cope with broken servers
+        not supporting implicit confirmation correctly.
+B<WARNING:> This setting leads to unspecified behavior and it is meant
+exclusively to allow interoperability with server implementations violating
+RFC 4210.
+
+=item B<OSSL_CMP_OPT_UNPROTECTED_SEND>
+
+        Send messages without CMP-level protection.
+
+=item B<OSSL_CMP_OPT_UNPROTECTED_ERRORS>
+
+        Accept unprotected error responses which are either explicitly
+        unprotected or where protection verification failed. Applies to regular
+        error messages as well as certificate responses (IP/CP/KUP) and
+        revocation responses (RP) with rejection.
+B<WARNING:> This setting leads to unspecified behavior and it is meant
+exclusively to allow interoperability with server implementations violating
+RFC 4210.
+
+=item B<OSSL_CMP_OPT_IGNORE_KEYUSAGE>
+
+        Ignore key usage restrictions in signer certificate when
+        validating signature-based protection in received CMP messages.
+        Else, 'digitalSignature' must be allowed by CMP signer certificates.
+
+=item B<OSSL_CMP_OPT_PERMIT_TA_IN_EXTRACERTS_FOR_IR>
+
+        Allow retrieving a trust anchor from extraCerts and using that
+        to validate the certificate chain of an IP message.
+
+=back
+
+OSSL_CMP_CTX_get_option() reads the current of the given option
```

Comment:
This does quite make sense

Reply:
Pleased to hear.

Reply:
Haha! That's funny! What I meant was "this *doesn't* quite make sense"!! The sentence seems to be missing a word...perhaps "reads the current *value* of the given option"